tflint --deep issues

I am trying to lint a piece of Terraform I have where I am specifying the AWS region in the provider block like:

```
provider "aws" {
  region = "us-west-2"
}
```

When running `tflint --deep`, I am getting the following error:

```
ERROR: error occurred in detecting. Please run with --debug option for details.
```

When running `tflint --deep --debug`, I am getting the following output:

```
DEBUG:/loader.go:120 [INFO] Load environment...
DEBUG:/loader.go:128 [ERROR] open .terraform/environment: no such file or directory
DEBUG:/loader.go:131 [INFO] Load tfstate...
DEBUG:/loader.go:145 [INFO] Local state detected
DEBUG:/loader.go:161 [INFO] Load tfvars...
DEBUG:/loader.go:164 [INFO] Load `terraform.tfvars`
DEBUG:/loader.go:166 [ERROR] stat terraform.tfvars: no such file or directory
DEBUG:/loader.go:191 [INFO] Load HCL file: `main.tf`
DEBUG:/loader.go:53 [INFO] Load HCL file: `main.tf`
DEBUG:/detector.go:153 [INFO] detect by `aws_instance_invalid_type`
DEBUG:/detector.go:153 [INFO] detect by `aws_instance_previous_type`
DEBUG:/detector.go:153 [INFO] detect by `aws_instance_not_specified_iam_profile`
DEBUG:/detector.go:153 [INFO] detect by `aws_instance_default_standard_volume`
DEBUG:/detector.go:153 [INFO] detect by `aws_instance_invalid_iam_profile`
DEBUG:/detector.go:153 [INFO] detect by `aws_instance_invalid_ami`
DEBUG:/aws_instance_invalid_ami.go:31 [ERROR] MissingRegion: could not find region configuration
DEBUG:/aws_instance_invalid_ami.go:48 [ERROR] ERROR: `"${data.aws_ami.ec2_ami.image_id}"` is not evaluable
DEBUG:/detector.go:153 [INFO] detect by `aws_instance_invalid_key_name`
DEBUG:/aws_instance_invalid_key_name.go:31 [ERROR] MissingRegion: could not find region configuration
DEBUG:/detector.go:153 [INFO] detect by `aws_instance_invalid_subnet`
DEBUG:/aws_instance_invalid_subnet.go:31 [ERROR] MissingRegion: could not find region configuration
DEBUG:/detector.go:153 [INFO] detect by `aws_instance_invalid_vpc_security_group`
DEBUG:/aws_instance_invalid_vpc_security_group.go:32 [ERROR] MissingRegion: could not find region configuration
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_alb_invalid_security_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_alb_invalid_subnet`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_alb_duplicate_name`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elb_invalid_security_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elb_invalid_subnet`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elb_invalid_instance`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elb_duplicate_name`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_default_parameter_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_invalid_vpc_security_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_invalid_db_subnet_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_invalid_parameter_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_invalid_option_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_invalid_type`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_previous_type`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_readable_password`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_db_instance_duplicate_identifier`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elasticache_cluster_default_parameter_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elasticache_cluster_invalid_parameter_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elasticache_cluster_invalid_subnet_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elasticache_cluster_invalid_security_group`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elasticache_cluster_invalid_type`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elasticache_cluster_previous_type`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_elasticache_cluster_duplicate_id`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_security_group_duplicate_name`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_invalid_route_table`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_not_specified_target`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_specified_multiple_targets`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_invalid_gateway`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_invalid_egress_only_gateway`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_invalid_nat_gateway`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_invalid_vpc_peering_connection`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_invalid_instance`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_route_invalid_network_interface`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_cloudwatch_metric_alarm_invalid_unit`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `aws_ecs_cluster_duplicate_name`
DEBUG:/detector.go:251 [INFO] targets are not found.
DEBUG:/detector.go:149 [INFO] skip `terraform_module_pinned_source`
ERROR: error occurred in detecting. Please run with --debug option for details.
```

--- a/client/aws.go
+++ b/client/aws.go
@@ -1,11 +1,10 @@
 package client
 
 import (
+	"errors"
 	"log"
+	"strings"
 
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/credentials"
-	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
 	"github.com/aws/aws-sdk-go/service/ecs"
@@ -20,6 +19,10 @@ import (
 	"github.com/aws/aws-sdk-go/service/iam/iamiface"
 	"github.com/aws/aws-sdk-go/service/rds"
 	"github.com/aws/aws-sdk-go/service/rds/rdsiface"
+	awsbase "github.com/hashicorp/aws-sdk-go-base"
+	"github.com/hashicorp/hcl2/gohcl"
+	"github.com/hashicorp/hcl2/hcl"
+	"github.com/hashicorp/terraform/configs"
 	homedir "github.com/mitchellh/go-homedir"
 )
 
@@ -52,13 +55,28 @@ type AwsCredentials struct {
 	Region    string
 }
 
+type awsProvider struct {
+	AccessKey     string `hcl:"access_key,optional"`
+	SecretKey     string `hcl:"secret_key,optional"`
+	Profile       string `hcl:"profile,optional"`
+	CredsFilename string `hcl:"shared_credentials_file,optional"`
+	Region        string `hcl:"region,optional"`
+
+	Remain hcl.Body `hcl:",remain"`
+}
+
 // NewAwsClient returns new AwsClient with configured session
-func NewAwsClient(creds AwsCredentials) (*AwsClient, error) {
+func NewAwsClient(provider *configs.Provider, creds AwsCredentials) (*AwsClient, error) {
 	log.Print("[INFO] Initialize AWS Client")
 
-	s, err := newAwsSession(creds)
+	config, err := getBaseConfig(provider, creds)
 	if err != nil {
 		return nil, err
+	}
+
+	s, err := awsbase.GetSession(config)
+	if err != nil {
+		return nil, formatBaseConfigError(err)
 	}
 
 	return &AwsClient{
@@ -72,44 +90,65 @@ func NewAwsClient(creds AwsCredentials) (*AwsClient, error) {
 	}, nil
 }
 
-// newAwsSession returns a session necessary for initialization of the AWS SDK
-func newAwsSession(creds AwsCredentials) (*session.Session, error) {
-	s := session.New()
+func getBaseConfig(provider *configs.Provider, creds AwsCredentials) (*awsbase.Config, error) {
+	base := &awsbase.Config{}
 
-	if creds.Region != "" {
-		log.Printf("[INFO] Set AWS region: %s", creds.Region)
-		s = session.New(&aws.Config{
-			Region: aws.String(creds.Region),
-		})
-	}
-	if creds.Profile != "" && creds.Region != "" {
-		var credsFile string
-		var err error
-		if creds.CredsFile != "" {
-			credsFile, err = homedir.Expand(creds.CredsFile)
-			if err != nil {
-				return nil, err
-			}
-		} else {
-			credsFile, err = homedir.Expand("~/.aws/credentials")
-			if err != nil {
-				// Maybe this is bug
-				panic(err)
-			}
+	if provider != nil {
+		pc, err := decodeProviderConfig(provider)
+		if err != nil {
+			return base, err
 		}
-		log.Printf("[INFO] Set AWS shared credentials: %s", credsFile)
-		s = session.New(&aws.Config{
-			Credentials: credentials.NewSharedCredentials(credsFile, creds.Profile),
-			Region:      aws.String(creds.Region),
-		})
-	}
-	if creds.AccessKey != "" && creds.SecretKey != "" && creds.Region != "" {
-		log.Printf("[INFO] Set AWS static credentials")
-		s = session.New(&aws.Config{
-			Credentials: credentials.NewStaticCredentials(creds.AccessKey, creds.SecretKey, ""),
-			Region:      aws.String(creds.Region),
-		})
+
+		base = &awsbase.Config{
+			AccessKey: pc.AccessKey,
+			Profile:   pc.Profile,
+			Region:    pc.Region,
+			SecretKey: pc.SecretKey,
+		}
+
+		path, err := homedir.Expand(pc.CredsFilename)
+		if err != nil {
+			return base, err
+		}
+		base.CredsFilename = path
 	}
 
-	return s, nil
+	if creds.AccessKey != "" {
+		base.AccessKey = creds.AccessKey
+	}
+	if creds.CredsFile != "" {
+		path, err := homedir.Expand(creds.CredsFile)
+		if err != nil {
+			return base, err
+		}
+		base.CredsFilename = path
+	}
+	if creds.Profile != "" {
+		base.Profile = creds.Profile
+	}
+	if creds.Region != "" {
+		base.Region = creds.Region
+	}
+	if creds.SecretKey != "" {
+		base.SecretKey = creds.SecretKey
+	}
+
+	return base, nil
+}
+
+func decodeProviderConfig(provider *configs.Provider) (awsProvider, error) {
+	var config awsProvider
+	diags := gohcl.DecodeBody(provider.Config, nil, &config)
+	if diags.HasErrors() {
+		return config, diags
+	}
+	return config, nil
+}
+
+// @see https://github.com/hashicorp/aws-sdk-go-base/blob/v0.3.0/session.go#L87
+func formatBaseConfigError(err error) error {
+	if strings.Contains(err.Error(), "No valid credential sources found for AWS Provider") {
+		return errors.New("No valid credential sources found")
+	}
+	return err
 }

--- a/client/aws_test.go
+++ b/client/aws_test.go
@@ -1,26 +1,33 @@
 package client
 
 import (
-	"reflect"
+	"os"
+	"path/filepath"
 	"testing"
 
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/google/go-cmp/cmp"
+	awsbase "github.com/hashicorp/aws-sdk-go-base"
+	"github.com/hashicorp/hcl2/hcl"
+	"github.com/hashicorp/hcl2/hclparse"
+	"github.com/hashicorp/terraform/configs"
 	homedir "github.com/mitchellh/go-homedir"
 )
 
-func Test_newAwsSession(t *testing.T) {
-	type Result struct {
-		Credentials *credentials.Credentials
-		Region      *string
+func Test_getBaseConfig(t *testing.T) {
+	currentDir, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
 	}
-	path, _ := homedir.Expand("~/.aws/credentials")
-	credPath, _ := homedir.Expand("~/.aws/creds")
+	home, err := homedir.Expand("~/")
+	if err != nil {
+		t.Fatal(err)
+	}
 
 	cases := []struct {
 		Name     string
 		Creds    AwsCredentials
-		Expected Result
+		File     string
+		Expected *awsbase.Config
 	}{
 		{
 			Name: "static credentials",
@@ -29,46 +36,110 @@ func Test_newAwsSession(t *testing.T) {
 				SecretKey: "AWS_SECRET_KEY",
 				Region:    "us-east-1",
 			},
-			Expected: Result{
-				Credentials: credentials.NewStaticCredentials("AWS_ACCESS_KEY", "AWS_SECRET_KEY", ""),
-				Region:      aws.String("us-east-1"),
+			Expected: &awsbase.Config{
+				AccessKey: "AWS_ACCESS_KEY",
+				SecretKey: "AWS_SECRET_KEY",
+				Region:    "us-east-1",
 			},
 		},
 		{
 			Name: "shared credentials",
 			Creds: AwsCredentials{
-				Profile: "production",
-				Region:  "us-east-1",
-			},
-			Expected: Result{
-				Credentials: credentials.NewSharedCredentials(path, "production"),
-				Region:      aws.String("us-east-1"),
-			},
-		},
-		{
-			Name: "shared credentials path",
-			Creds: AwsCredentials{
 				Profile:   "default",
 				CredsFile: "~/.aws/creds",
 				Region:    "us-east-1",
 			},
-			Expected: Result{
-				Credentials: credentials.NewSharedCredentials(credPath, "default"),
-				Region:      aws.String("us-east-1"),
+			Expected: &awsbase.Config{
+				Profile:       "default",
+				CredsFilename: filepath.Join(home, ".aws", "creds"),
+				Region:        "us-east-1",
+			},
+		},
+		{
+			Name: "static credentials provider",
+			File: "static-creds.tf",
+			Expected: &awsbase.Config{
+				AccessKey: "AWS_ACCESS_KEY",
+				SecretKey: "AWS_SECRET_KEY",
+				Region:    "us-east-1",
+			},
+		},
+		{
+			Name: "shared credentials provider",
+			File: "shared-creds.tf",
+			Expected: &awsbase.Config{
+				Profile:       "default",
+				CredsFilename: filepath.Join(home, ".aws", "creds"),
+				Region:        "us-east-1",
+			},
+		},
+		{
+			Name:     "assume role provider",
+			File:     "assume-role.tf",
+			Expected: &awsbase.Config{},
+		},
+		{
+			Name: "prefer tflint static credentials over provider",
+			File: "static-creds.tf",
+			Creds: AwsCredentials{
+				AccessKey: "TFLINT_AWS_ACCESS_KEY",
+				SecretKey: "TFLINT_AWS_SECRET_KEY",
+				Region:    "us-east-2",
+			},
+			Expected: &awsbase.Config{
+				AccessKey: "TFLINT_AWS_ACCESS_KEY",
+				SecretKey: "TFLINT_AWS_SECRET_KEY",
+				Region:    "us-east-2",
+			},
+		},
+		{
+			Name: "prefer tflint shared credentials over provider",
+			File: "shared-creds.tf",
+			Creds: AwsCredentials{
+				Profile:   "terraform",
+				CredsFile: "~/.aws/tf_credentials",
+				Region:    "us-east-2",
+			},
+			Expected: &awsbase.Config{
+				Profile:       "terraform",
+				CredsFilename: filepath.Join(home, ".aws", "tf_credentials"),
+				Region:        "us-east-2",
 			},
 		},
 	}
 
 	for _, tc := range cases {
-		s, err := newAwsSession(tc.Creds)
+		var pc *configs.Provider
+		if tc.File != "" {
+			parser := hclparse.NewParser()
+			f, diags := parser.ParseHCLFile(filepath.Join(currentDir, "test-fixtures", tc.File))
+			if diags.HasErrors() {
+				t.Fatal(diags)
+			}
+
+			body, _, diags := f.Body.PartialContent(&hcl.BodySchema{
+				Blocks: []hcl.BlockHeaderSchema{
+					{
+						Type:       "provider",
+						LabelNames: []string{"name"},
+					},
+				},
+			})
+			if diags.HasErrors() {
+				t.Fatal(diags)
+			}
+
+			pc = &configs.Provider{
+				Config: body.Blocks[0].Body,
+			}
+		}
+
+		base, err := getBaseConfig(pc, tc.Creds)
 		if err != nil {
 			t.Fatalf("Failed `%s` test: Unexpected error occurred: %s", tc.Name, err)
 		}
-		if !reflect.DeepEqual(tc.Expected.Credentials, s.Config.Credentials) {
-			t.Fatalf("Failed `%s` test: expected credentials are `%#v`, but get `%#v`", tc.Name, tc.Expected.Credentials, s.Config.Credentials)
-		}
-		if !reflect.DeepEqual(tc.Expected.Region, s.Config.Region) {
-			t.Fatalf("Failed `%s` test: expected region are `%#v`, but get `%#v`", tc.Name, tc.Expected.Region, s.Config.Region)
+		if !cmp.Equal(tc.Expected, base) {
+			t.Fatalf("Failed `%s` test: Diff=%s", tc.Name, cmp.Diff(tc.Expected, base))
 		}
 	}
 }

--- a/client/test-fixtures/assume-role.tf
+++ b/client/test-fixtures/assume-role.tf
@@ -1,0 +1,7 @@
+provider "aws" {
+  assume_role {
+    role_arn     = "arn:aws:iam::ACCOUNT_ID:role/ROLE_NAME"
+    session_name = "SESSION_NAME"
+    external_id  = "EXTERNAL_ID"
+  }
+}

--- a/client/test-fixtures/shared-creds.tf
+++ b/client/test-fixtures/shared-creds.tf
@@ -1,0 +1,5 @@
+provider "aws" {
+    profile                 = "default"
+    shared_credentials_file = "~/.aws/creds"
+    region                  = "us-east-1"
+}

--- a/client/test-fixtures/static-creds.tf
+++ b/client/test-fixtures/static-creds.tf
@@ -1,0 +1,5 @@
+provider "aws" {
+    access_key = "AWS_ACCESS_KEY"
+    secret_key = "AWS_SECRET_KEY"
+    region     = "us-east-1"
+}

--- a/docs/guides/credentials.md
+++ b/docs/guides/credentials.md
@@ -5,11 +5,14 @@ In [Deep checking](advanced.md#deep-checking), it is necessary to set provider's
 Credentials are used with the following priority:
 
 - Static credentials
+- Static credentials (Terraform)
+- Environment variables
 - Shared credentials
-- Environment credentials
-- Default shared credentials
+- Shared credentials (Terraform)
+- ECS and CodeBuild task roles
+- EC2 role
 
-## Static Credentials
+## Static credentials
 
 If you have an access key and a secret key, you can pass these keys like the following:
 
@@ -27,7 +30,17 @@ config {
 }
 ```
 
-## Shared Credentials
+Although there is not recommended, if an access key is hard-coded in a provider definition, they will also be taken into account. However, aliases are not supported. The priority is higher than the environment variable and lower than the above way.
+
+```hcl
+provider "aws" {
+  region     = "us-west-2"
+  access_key = "my-access-key"
+  secret_key = "my-secret-key"
+}
+```
+
+## Shared credentials
 
 If you have [shared credentials](https://aws.amazon.com/jp/blogs/security/a-new-and-standardized-way-to-manage-credentials-in-the-aws-sdks/), you can pass a profile name and credentials file path. If omitted, these will be `default` and `~/.aws/credentials`.
 
@@ -45,7 +58,17 @@ config {
 }
 ```
 
-## Environment Credentials
+If these configurations are defined in the provider block, they will also be taken into account. But the priority is lower than the above way.
+
+```hcl
+provider "aws" {
+  region                  = "us-west-2"
+  shared_credentials_file = "/Users/tf_user/.aws/creds"
+  profile                 = "customprofile"
+}
+```
+
+## Environment variables
 
 TFLint looks up `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`, `AWS_REGION` environment variables. This is useful when you don't want to explicitly pass credentials.
 
@@ -53,3 +76,7 @@ TFLint looks up `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`, `AWS_REGION` en
 $ export AWS_ACCESS_KEY_ID=AWS_ACCESS_KEY
 $ export AWS_SECRET_ACCESS_KEY=AWS_SECRET_KEY
 ```
+
+## Role-based authentication
+
+TFLint fetches AWS credentials in the same way as Terraform. See [this documentation](https://www.terraform.io/docs/providers/aws/index.html#ecs-and-codebuild-task-roles) for role-based authentication. However, Assume role is not supported.

--- a/go.mod
+++ b/go.mod
@@ -7,6 +7,7 @@ require (
 	github.com/fatih/color v1.7.0
 	github.com/golang/mock v1.3.1
 	github.com/google/go-cmp v0.3.0
+	github.com/hashicorp/aws-sdk-go-base v0.3.0
 	github.com/hashicorp/go-version v1.2.0
 	github.com/hashicorp/hcl v0.0.0-20180404174102-ef8a98b0bbce // indirect
 	github.com/hashicorp/hcl2 v0.0.0-20190702185634-5b39d9ff3a9a

--- a/tflint/runner.go
+++ b/tflint/runner.go
@@ -50,9 +50,14 @@ func NewRunner(c *Config, ants map[string]Annotations, cfg *configs.Config, vari
 	}
 	log.Printf("[INFO] Initialize new runner for %s", path)
 
-	awsClient, err := client.NewAwsClient(c.AwsCredentials)
-	if err != nil {
-		return nil, err
+	awsClient := &client.AwsClient{}
+	var err error
+	if c.DeepCheck {
+		// FIXME: Alias providers are not considered
+		awsClient, err = client.NewAwsClient(cfg.Module.ProviderConfigs["aws"], c.AwsCredentials)
+		if err != nil {
+			return nil, err
+		}
 	}
 
 	return &Runner{
@@ -148,6 +153,8 @@ func NewModuleRunners(parent *Runner) ([]*Runner, error) {
 		if err != nil {
 			return runners, err
 		}
+		// Inherit parent's AwsClient
+		runner.AwsClient = parent.AwsClient
 		runners = append(runners, runner)
 		moudleRunners, err := NewModuleRunners(runner)
 		if err != nil {

